pkg/cloud/powervs.go

```diff
@@ -252,3 +256,35 @@ func (p *powerVSCloud) GetDiskByID(volumeID string) (disk *Disk, err error) {
 		CapacityGiB: int64(*v.Size),
 	}, nil
 }
+
+func readCredentials() (string, error) {
+	apiKey, err := readCredentialsFromFile()
+	if err != nil {
+		return "", err
+	}
+	if apiKey != "" {
+		return apiKey, nil
+	}
+
+	klog.Info("Falling back to read IBMCLOUD_API_KEY environment variable for the key")
+	apiKey = os.Getenv("IBMCLOUD_API_KEY")
+	if apiKey == "" {
+		return "", fmt.Errorf("IBMCLOUD_API_KEY is not provided")
```

Comment:
nit: 
We could switch to errors.New as the error string is static + has minor gains in op time. We could take track this in a separate issue, altogether. 


```suggestion
		return "", errors.New("IBMCLOUD_API_KEY is not provided")
```

Reply:
https://github.com/kubernetes-sigs/ibm-powervs-block-csi-driver/issues/755 - created a GHE to track this